buckygen FTBFS with Apple clang version 21.0.0

https://github.com/passagemath/passagemath/actions/runs/28553198860/job/84677115934#step:14:14359
```
  [buckygen-1.1]   [spkg-install] gcc -std=gnu23 -g -O2 -O3 buckygen.c -o buckygen
  [buckygen-1.1]   [spkg-install] In file included from buckygen.c:280:
  [buckygen-1.1]   [spkg-install] ./splay.c:260:14: error: too many arguments to function call, expected 0, have 2
  [buckygen-1.1]   [spkg-install]   260 |             PRESENT(p);
  [buckygen-1.1]   [spkg-install]       |             ~~~~~~~~^~
  [buckygen-1.1]   [spkg-install] buckygen.c:274:34: note: expanded from macro 'PRESENT'
  [buckygen-1.1]   [spkg-install]   274 | #define PRESENT(p) old_splaynode(p, is_new_node)
  [buckygen-1.1]   [spkg-install]       |                    ~~~~~~~~~~~~~ ^~~~~~~~~~~~~~
  [buckygen-1.1]   [spkg-install] buckygen.c:278:6: note: 'old_splaynode' declared here
  [buckygen-1.1]   [spkg-install]   278 | void old_splaynode();
  [buckygen-1.1]   [spkg-install]       |      ^
  [buckygen-1.1]   [spkg-install] In file included from buckygen.c:280:
  [buckygen-1.1]   [spkg-install] ./splay.c:283:17: error: too many arguments to function call, expected 0, have 5
  [buckygen-1.1]   [spkg-install]   283 |     NOT_PRESENT(new_node);
  [buckygen-1.1]   [spkg-install]       |     ~~~~~~~~~~~~^~~~~~~~~
```

--- a/build/pkgs/buckygen/spkg-install.in
+++ b/build/pkgs/buckygen/spkg-install.in
@@ -6,6 +6,8 @@ fi
 
 cd src
 
+export CFLAGS="$CFLAGS -std=gnu17"
+
 $MAKE
 if [ $? -ne 0 ]; then
     echo >&2 "Error building buckygen."

